### DISCUSSION TITLE
Issue with grpc based exporters

### DISCUSSION THREAD

**Discussion by shivchelwa** [open-telemetry/opentelemetry-java#2478] [Q&A] [answered]
Hi,

I just started exploring manual instrumentation for my java application. When I tried to create a Tracer using below code, it fails as one of the grpc dependency is not found:

```
	SpanExporter spanExporter =  OtlpGrpcSpanExporter.builder().readProperties(properties).build();
    	SpanProcessor spanProcessor = BatchSpanProcessor.builder(spanExporter).build();
    	OpenTelemetrySdk.getGlobalTracerManagement().addSpanProcessor(spanProcessor);
        Tracer tracer = GlobalOpenTelemetry.get().getTracerProvider().get(MyClass.class.getPackageName());
```

The below exception is thrown:

> Exception in thread "main" io.grpc.ManagedChannelProvider$ProviderNotFoundException: No functional channel service provider found. Try adding a dependency on the grpc-okhttp, grpc-netty, or grpc-netty-shaded artifact
>	at io.grpc.ManagedChannelProvider.provider(ManagedChannelProvider.java:65)
>	at io.grpc.ManagedChannelBuilder.forTarget(ManagedChannelBuilder.java:76)
>	at io.opentelemetry.exporter.jaeger.JaegerGrpcSpanExporterBuilder.build(JaegerGrpcSpanExporterBuilder.java:97)
>      ...

Shouldn't grpc-okhttp, grpc-netty, or grpc-netty-shaded be one of the dependencies of the exporter(jaeger/otlp) project?

When I added this dependency explicitly and I get a different error most likely due to version compatibility. What version of grpc-okhttp should be added to resolve this issue?

**Comment by jkwatson** [comment, marked as answer]
Hi @shivchelwa . We specifically don't include one of the grpc channel providers because people may want to bring their own. Our examples use `grpc-netty-shaded`, so we know that works, at least with compatible versions.

**Reply by anuraaga** [reply]
@shivchelwa You will also need to use `grpc-bom` to manage grpc dependency conflicts, I've found there's generally a conflict unless adding the bom. If you use the bom, you should be able to use most versions of gRPC, I haven't dug into what our minimum would be but we don't use any special APIs so should be using pretty old stuff.
